pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -2541,7 +2541,25 @@ public boolean isWrapperFor(Class<?> iface) throws SQLException {
 
   public ResultSet getFunctions(String catalog, String schemaPattern, String functionNamePattern)
       throws SQLException {
-    return getProcedures(catalog, schemaPattern, functionNamePattern);
+    String sql;
+    sql = "SELECT NULL AS FUNCTION_CAT, n.nspname AS FUNCTION_SCHEM, p.proname AS FUNCTION_NAME, "
+            + "d.description AS REMARKS, "
+            + DatabaseMetaData.procedureReturnsResult + " AS FUNCTION_TYPE, "
```

Comment:
Welcome to the ambiguity of JDBC. I would say we are free to implement it as we see fit and then argue later with people who see it differently. I would say rowtype(table) returns true. refcursor, false.